build/Shared/StringBuilderPool.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NuGet
+{
+    /// <summary>
+    /// Provides a resource pool that enables reusing instances of <see cref="StringBuilder"/> instances.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Renting and returning buffers with an <see cref="StringBuilderPool"/> can increase performance
+    /// in situations where <see cref="StringBuilder"/> instances are created and destroyed frequently,
+    /// resulting in significant memory pressure on the garbage collector.
+    /// </para>
+    /// <para>
+    /// This class is thread-safe.  All members may be used by multiple threads concurrently.
+    /// </para>
+    /// </remarks>
+    internal class StringBuilderPool
```

Comment:
Please add a private constructor like `ArrayPool` to avoid creating instances of this type. I am okay if this is done in the follow-up PR.

https://github.com/NuGet/NuGet.Client/blob/f04b570e65af9f7828f01968c1efdf42e8f317d7/src/NuGet.Core/NuGet.Packaging/ArrayPool.cs#L33-L35

Reply:
https://github.com/NuGet/NuGet.Client/pull/4146